File: exercicios_2_C/14_fn_fatorial.c
/*
14) Escreva uma função que calcule e retorne o fatorial de um número fornecido pelo usuário.
*/ 
#include <stdio.h> // Inclui a biblioteca padrão de entrada e saída.

// Função para calcular o fatorial de um número inteiro.
int calcularFatorial(int numero) {
    if (numero == 0) { // Verifica se o número é igual a 0.
        return 1; // Se for, retorna 1 (definição do fatorial de 0).
    } else { // Caso contrário.
        int fatorial = 1; // Inicializa a variável para armazenar o fatorial como 1.

        // Loop para calcular o fatorial do número.
        for (int i = 1; i <= numero; i++) {
            fatorial *= i; // Multiplica o fatorial pela variável 'i'.
        }
        return fatorial; // Retorna o resultado do fatorial.
    }
}

int main() { // Função principal.
    int num, resultado; // Variáveis para armazenar o número fornecido pelo usuário e o resultado do cálculo.

    printf("Digite um número inteiro: "); // Solicita ao usuário que digite um número inteiro.
    scanf("%d", &num); // Lê e armazena o número fornecido pelo usuário.

    resultado = calcularFatorial(num); // Chama a função 'calcularFatorial' para calcular o fatorial do número fornecido.

    printf("O fatorial de %d é: %d\n", num, resultado); // Imprime o fatorial do número fornecido.

    return 0; // Indica que o programa foi executado com sucesso.
}

/*
saída:
Digite um número inteiro: 2
O fatorial de 2 é: 2
*/ 

File: exercicios_2_Rust/14_fn_fatorial.rs
/*
14) Escreva uma função que calcule e retorne o fatorial de um número fornecido pelo usuário.
*/ 
use std::io;

fn calcular_fatorial(numero: u64) -> u64 {

    if numero == 0 {
        return 1;
    } else {
        let mut fatorial = 1;

        for i in 1..=numero {
            fatorial *= i;
        }
        return fatorial;
    }
}

fn main() {
    let mut input = String::new();

    println!("Digite um número inteiro: ");
    io::stdin().read_line(&mut input).expect("Falha ao ler a entrada");
    let num: u64 = input.trim().parse().expect("Entrada inválida");

    let resultado = calcular_fatorial(num);

    println!("O fatorial de {} é: {}", num, resultado);
}

/*
saída:
Digite um número inteiro: 2
O fatorial de 2 é: 2
*/ 